From 809ec73924b899946310323c58327c0ac2ebfe7b Mon Sep 17 00:00:00 2001
From: Masahiro Yamada <masahiroy@kernel.org>
Date: Mon, 1 Jun 2020 14:57:21 +0900
Subject: [PATCH] modpost: remove get_next_text() and make {grab,release_}file
 static

Patch-mainline: No, it reverts part of mainline code.
References: build#SUSE-Euler-2.0

get_next_line() is no longer used. Remove.

grab_file() and release_file() are only used in modpost.c. Make them
static.

Signed-off-by: Masahiro Yamada <masahiroy@kernel.org>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 scripts/mod/modpost.c | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index 352392b5319e..7aed460cccd7 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -477,6 +477,40 @@ static void *grab_file(const char *filename, size_t *size)
 	return map;
 }
 
+/**
+  * Return a copy of the next line in a mmap'ed file.
+  * spaces in the beginning of the line is trimmed away.
+  * Return a pointer to a static buffer.
+  **/
+static char *get_next_line(unsigned long *pos, void *file, unsigned long size)
+{
+	static char line[4096];
+	int skip = 1;
+	size_t len = 0;
+	signed char *p = (signed char *)file + *pos;
+	char *s = line;
+
+	for (; *pos < size ; (*pos)++) {
+		if (skip && isspace(*p)) {
+			p++;
+			continue;
+		}
+		skip = 0;
+		if (*p != '\n' && (*pos < size)) {
+			len++;
+			*s++ = *p++;
+			if (len > 4095)
+				break; /* Too long, stop */
+		} else {
+			/* End of string */
+			*s = '\0';
+			return line;
+		}
+	}
+	/* End of buffer */
+	return NULL;
+}
+
 static void release_file(void *file, size_t size)
 {
 	munmap(file, size);
-- 
2.26.2

